spring-boot-admin-client/src/main/java/de/codecentric/boot/admin/client/registration/DefaultApplicationFactory.java

```diff
@@ -192,11 +210,31 @@ protected String getManagementHost() {
 	}
 
 	protected InetAddress getLocalHost() {
+		// Try using Spring Cloud Commons InetUtils if available (optional dependency)
+		if (this.inetUtils != null && this.inetUtils.isPresent()) {
+			Object utils = this.inetUtils.get();
+			try {
+				java.lang.reflect.Method m = utils.getClass().getMethod("findFirstNonLoopbackHostInfo");
+				Object host = m.invoke(utils);
+				if (host instanceof String && StringUtils.hasText((String) host)) {
```

Comment:
It can probably be simplified with pattern matching for `instanceof` if the Java version of the project allows it ;)